scripts/unreleased-change-monitor.ts

```diff
@@ -0,0 +1,124 @@
+import { Octokit } from 'octokit';
+import { LimaAndQemu, AlpineLimaISO } from 'scripts/dependencies/lima';
+import * as tools from 'scripts/dependencies/tools';
+import { WSLDistro, HostResolverHost } from 'scripts/dependencies/wsl';
+import { Dependency, UnreleasedChangeMonitor, getOctokit } from 'scripts/lib/dependencies';
+
+const GITHUB_OWNER = process.env.GITHUB_REPOSITORY?.split('/')[0] || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+// a (hopefully) unique and communicative key that is used to find issues created by
+// this script by filtering them down to the ones that have it in their title
+const UCMONITOR = 'ucmonitor';
+
+type UnreleasedChangeMonitoringDependency = Dependency & UnreleasedChangeMonitor;
+
+type DependencyState = {
+  dependency: UnreleasedChangeMonitoringDependency,
+  latestReleaseTag: string,
+  hasUnreleasedChanges: boolean,
+};
+
+const dependencies: UnreleasedChangeMonitoringDependency[] = [
```

Comment:
It would be good if we could centralize the list of dependencies, and check all dependencies if they implement the desired interface. But that can be later.